pkg/kubectl/cmd/delete.go

```diff
@@ -234,12 +243,16 @@ func (o *DeleteOptions) RunDelete() error {
 	shortOutput := o.Output == "name"
 	// By default use a reaper to delete all related resources.
 	if o.Cascade {
-		return ReapResult(o.Result, o.f, o.Out, true, o.IgnoreNotFound, o.Timeout, o.GracePeriod, o.WaitForDeletion, shortOutput, o.Mapper, false)
+		return ReapResultWithWaitingOptions(o.Result, o.f, o.Out, true, o.IgnoreNotFound, o.Timeout, o.GracePeriod, o.WaitForGracefulDeletion, shortOutput, o.Mapper, false, o.WaitForDeletion, o.IsDefaultWaitForDeletion)
 	}
 	return DeleteResult(o.Result, o.Out, o.IgnoreNotFound, shortOutput, o.Mapper)
```

Comment:
What happens if --cascade=false but --wait=true? Does DeleteResult guarantee the object is deleted before returning?